lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -601,28 +611,26 @@ export default class BitcoinProcessor {
     const endBlockHeight = await this.bitcoinClient.getCurrentBlockHeight();
     console.info(`Processing transactions from ${startBlockHeight} to ${endBlockHeight}`);
 
+    let blockHeight = startBlockHeight;
     let previousBlockHash = startBlock.previousHash;
+    while (blockHeight <= endBlockHeight) {
+      const processedBlockMetadata = await this.processBlock(blockHeight, previousBlockHash);
 
-    for (let blockHeight = startBlockHeight; blockHeight <= endBlockHeight; blockHeight++) {
-      const processedBlockHash = await this.processBlock(blockHeight, previousBlockHash);
+      await this.blockMetadataStore.add([processedBlockMetadata]);
 
-      this.lastProcessedBlock = {
-        height: blockHeight,
-        hash: processedBlockHash,
-        previousHash: previousBlockHash
-      };
+      this.lastProcessedBlock = processedBlockMetadata;
```

Comment:
No explicit cast needed? This probably works because the property names are same? If that's the reason then maybe we can initialize the `this.lastProcessedBlock` explicitly in case the property names change in the future?

Another option might be to change the type of `this.lastProcessedBlock` to `BlockMetadata` but that might trigger more cascading changes (especially in the test) so ok with not doing that.

Reply:
It is also my hope that long term we eliminate `IBlockInfo` and use just `BlockMetadata`, but as you said it has more cascading effect and bloats the PR even further.

In the meantime we could make `BlockMetadata` extend `IBlockInfo` (which is what I did prior to this PR), but I didn't want to take `IBlockInfo` out of this file.

All is to say, I will explicitly set them as you've suggested for now.